### PR TITLE
Prefill workspace invitation email (#7174)

### LINKED ISSUE #7174: Follow-ups after new email invite feature
### Issue 1 (design): Alignment

Screenshot 2024-09-20 at 09 34 19

### Issue 2 (nice to have): email prefill or don't check email

Screenshot 2024-09-20 at 09 28 38

If you already know my email then you should prefill it in the UI!
But do we really want to force the email to be the same? Security vs UX tradeoff - what is the market practise? (Check Notion, Linear, etc.). 
cc @Bonapara any opinion?

###  Issue 3 (bug): Message when accepting the invite

This is not a blocking error but I was showing this when accepting the invite

Screenshot 2024-09-20 at 09 29 11

### PR DESCRIPTION
**Pull request by eliasylonen**
Prefill workspace invitation email, fixes #7174

### CODE REVIEW

File: packages/twenty-front/src/modules/client-config/components/ClientConfigProviderEffect.tsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ export const ClientConfigProviderEffect = () => {
   const setIsDebugMode = useSetRecoilState(isDebugModeState);
   const setIsAnalyticsEnabled = useSetRecoilState(isAnalyticsEnabledState);
 
-  const setIsSignInPrefilled = useSetRecoilState(isSignInPrefilledState);
+  const setisDeveloperDefaultSignInPrefilled = useSetRecoilState(
```
Comment (AMoreaux) [inline packages/twenty-front/src/modules/client-config/components/ClientConfigProviderEffect.tsx:23]:
```suggestion
  const setIsDeveloperDefaultSignInPrefilled = useSetRecoilState(
```

File: packages/twenty-front/src/modules/client-config/components/ClientConfigProviderEffect.tsx [hunk 2]
```diff
@@ -76,7 +78,7 @@ export const ClientConfigProviderEffect = () => {
     });
     setIsDebugMode(data?.clientConfig.debugMode);
     setIsAnalyticsEnabled(data?.clientConfig.analyticsEnabled);
-    setIsSignInPrefilled(data?.clientConfig.signInPrefilled);
+    setisDeveloperDefaultSignInPrefilled(data?.clientConfig.signInPrefilled);
```
Comment (AMoreaux) [inline packages/twenty-front/src/modules/client-config/components/ClientConfigProviderEffect.tsx:81]:
```suggestion
    setIsDeveloperDefaultSignInPrefilled(data?.clientConfig.signInPrefilled);
```

**Review by AMoreaux — APPROVED** [review]
Nice UX improvement! 🙏
I noticed a small typo, and as I mentioned in the issue, the last point remains during my test."

**Comment by FelixMalfait** [comment]
Geat! How are would it be to add a matching step like we do in the Select field? I think it would be nice for it to be consistent for the two field types?
Screenshot 2024-12-02 at 10 57 06

**Comment by FelixMalfait** [comment]
I think the error might have to do with clicking twice on the button? If it's too slow users tend to click twice. Could be a wrong lead. Let's close the PR like this and you can fix the item (3) in another PR @eliasylonen?

**Pull request merged**